sdk/healthinsights/azure-health-insights-radiologyinsights/src/main/java/com/azure/health/insights/radiologyinsights/models/FhirR4DomainResource.java

```diff
@@ -63,9 +50,9 @@ public class FhirR4DomainResource extends FhirR4Resource {
      * @param resourceType the resourceType value to set.
      */
     @Generated
-    @JsonCreator
-    public FhirR4DomainResource(@JsonProperty(value = "resourceType") String resourceType) {
+    public FhirR4DomainResource(String resourceType) {
         super(resourceType);
+        setResourceType("Fhir_R4_DomainResource");
```

Comment:
`setResourceType` not generated in class.

The class is a bit unusually. It has a base class of `FhirR4Resource`, but only this class have discriminator property.

Reply:
Thanks for finding this. Looks like there is an edge case with code generation where the discriminator is a required property.